pkg/asset/agent/image/ignition.go

```diff
@@ -411,10 +411,12 @@ func getRendezvousHostEnv(serviceProtocol, nodeZeroIP string, workflowType workf
 	return fmt.Sprintf(`NODE_ZERO_IP=%s
 SERVICE_BASE_URL=%s
 IMAGE_SERVICE_BASE_URL=%s
+PULL_SECRET_TOKEN=%s
```

Comment:
This currently results in the same token being used for both the agents and the scripts that drive the assisted API. I can't remember what we said about how we would do authz in the future, but maybe it makes sense to pass them separately even if they have the same value right now.
I guess that would also solve the issue that "PULL_SECRET_TOKEN" is [not the ideal name](https://github.com/openshift/assisted-service/pull/6174#discussion_r1598582156) from assisted-service's perspective, bit is required for assisted-installer-agent.

Reply:
yes, its been renamed to AGENT_AUTH_TOKEN

Reply:
Is PULL_SECRET_TOKEN needed? I'm having trouble understanding how it is used.

Reply:
The assisted-installer-agent needs `PULL_SECRET_TOKEN` for the authenticator to work else throws a warning `Host will stop trying to register; user is not authenticated to perform host registration` and the cluster installation waits forever.

https://github.com/openshift/assisted-installer-agent/blob/10f41e28048f31179046c8af78c63d4454cfc340/src/commands/register_node.go#L17-L45

Reply:
Let's add a comment to explain why PULL_SECRET_TOKEN is here and how it is used in assisted-installer-agent.

Reply:
Done